include/sys/zfs_sa.h

```diff
@@ -87,6 +88,8 @@ typedef enum zpl_attr {
 #define	SA_UID_OFFSET		24
 #define	SA_GID_OFFSET		32
 #define	SA_PARENT_OFFSET	40
+#define	SA_FLAGS_OFFSET		48
+#define	SA_PROJID_OFFSET	128
```

Comment:
> Follows current user/group quota framework, I have no better idea yet

I think you're saying that you can't follow my suggestion of putting PROJID before RDEV.  How does that conflict with the "current user/group quota framework"?

But it looks like you've made the change I suggested, so maybe I've misunderstood?  You've put PROJID between LINKS and RDEV.  Is its OFFSET still 128 (I don't think so, but I don't see how your code could work if SA_PROJID_OFFSET is inaccurate).

I think we should be aware that there can be exactly one "optional, but fixed location" attribute, and it is PROJID.  If another feature wants to do something similar to what you've done, they won't be able to.